### ISSUE TITLE
`TwoPaneView` has infinite height

### ISSUE THREAD

**Issue by morning4coffe-dev** [unoplatform/uno#14133]
### Current behavior

When using the `TwoPaneView` control with GridView I noticed my button that is supposed to be always visible is rendered on the bottom on Uno platforms which should not happen and this behavior differs from WinUI.

This is how it currently looks on Android, notice I had to scroll down to see the Button

![image](https://github.com/unoplatform/uno/assets/78549750/6cd8abbd-faae-4c57-8f58-c243000a6c31)

### Expected behavior

On Windows the Button is always visible on the screen

![image](https://github.com/unoplatform/uno/assets/78549750/90fed4a1-d386-4ab8-8c29-da2772948bcb)

### How to reproduce it (as minimally and precisely as possible)

https://github.com/morning4coffe-dev/uno-platform-testing/tree/doti/twopaneview-height

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### Affected platforms

WebAssembly, Android, iOS, macOS (AppKit), Mac Catalyst, Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer)

**Comment by morning4coffe-dev** [comment]
CC @MartinZikmund

**Comment by Youssef1313** [comment]
Have you tested Skia/Wasm?

**Comment by morning4coffe-dev** [comment]
@Youssef1313 Yes, the issue is very similar there, but it seems like it is even a little worse as the GridView stretches the height of its items too

**Comment by Youssef1313** [comment]
I just tested and it's indeed very broken on Skia. I'll investigate.

**Comment by Youssef1313** [comment]
`ItemsWrapGrid` is actually not implemented on Skia.

**Comment by Youssef1313** [comment]
Skia part (**not including ItemsWrapGrid support**) is being done in #14147.

**Comment by MartinZikmund** [comment]
Repro branch created: \`dev/mazi/repro/14133\`

**Confirmed on Skia Desktop.** Two runtime tests were added to \`Given_TwoPaneView\`, both failing as expected:

1. \`When_TwoPaneView_In_Finite_Container_Pane1_Has_Finite_Height\` — ListView inside a 400px TwoPaneView measures 4000px tall (100 items × 40px default height), indicating TwoPaneView passes an unconstrained/infinite height to its panes.

2. \`When_TwoPaneView_Pane1_Sibling_Button_Is_Visible\` — A Button placed below a ListView in Pane1's StackPanel is positioned at Y=4096px, well outside the TwoPaneView's 466px bounds.

**Root cause**: TwoPaneView does not constrain the height of its pane content during measure, passing an infinite available height instead of its own finite height.
